server/src/queries.rs

```diff
@@ -0,0 +1,252 @@
+        if !is_aircraft_available || !aircraft_flights.is_empty() {
+            return Err(Status::not_found("Aircraft not available"));
+        }
+    }
+    //6. TODO: check other constraints (cargo weight, number of passenger seats)
+    //7. create draft flight plan (in memory)
+    let fp_id = Uuid::new_v4().to_string();
+    unconfirmed_flight_plans().lock().unwrap().insert(
+        fp_id.clone(),
+        FlightPlan {
+            id: 1234, //todo string id
+            flight_status: FlightStatus::Ready as i32,
+        },
+    );
+    //8. automatically cancel draft flight plan if not confirmed by user
+    cancel_flight_after_timeout(fp_id.clone());
+    //9. return response - TODO copy from storage flight plan
+    let item = QueryFlightPlan {
+        id: fp_id,
+        pilot_id: 1,
+        aircraft_id: 1,
+        cargo: [123].to_vec(),
+        weather_conditions: "Sunny, no wind :)".to_string(),
+        vertiport_id_departure: 1,
+        pad_id_departure: 1,
+        vertiport_id_destination: 1,
+        pad_id_destination: 1,
+        estimated_departure: Some(Timestamp {
+            seconds: departure_time.timestamp(),
+            nanos: departure_time.timestamp_subsec_nanos() as i32,
+        }),
+        estimated_arrival: Some(Timestamp {
+            seconds: arrival_time.timestamp(),
+            nanos: arrival_time.timestamp_subsec_nanos() as i32,
+        }),
+        actual_departure: None,
+        actual_arrival: None,
+        flight_release_approval: None,
+        flight_plan_submitted: None,
+        flight_status: FlightStatus::Ready as i32,
+        flight_priority: FlightPriority::Low as i32,
+    };
+    let response = QueryFlightResponse {
+        flights: [item].to_vec(),
+    };
+    Ok(Response::new(response))
+}
+
+fn get_fp_by_id(id: String) -> Option<FlightPlan> {
+    unconfirmed_flight_plans().lock().unwrap().get(&id).copied()
+}
+
+///Confirms the flight plan
+pub async fn confirm_flight(
+    request: Request<Id>,
+    mut storage_client: StorageRpcClient<tonic::transport::Channel>,
+) -> Result<Response<ConfirmFlightResponse>, Status> {
+    let fp_id = request.into_inner().id;
+    let draft_fp = get_fp_by_id(fp_id.clone());
+    return if draft_fp.is_none() {
+        Err(Status::not_found("Flight plan not found"))
+    } else {
+        let fp = storage_client
+            .insert_flight_plan(Request::new(draft_fp.unwrap()))
+            .await?
+            .into_inner();
+        let sys_time = SystemTime::now();
+        let response = ConfirmFlightResponse {
+            id: fp.id.to_string(), //todo this should be string
+            confirmed: true,
+            confirmation_time: Some(Timestamp::from(sys_time)),
+        };
+        unconfirmed_flight_plans().lock().unwrap().remove(&fp_id);
+        Ok(Response::new(response))
+    };
+}
+
+/// Cancels a draft flight plan
+pub async fn cancel_flight(request: Request<Id>) -> Result<Response<CancelFlightResponse>, Status> {
```

Comment:
Cancellations can occur on any flight plan, not just drafts. Creating an issue for this